### DISCUSSION TITLE
Help! Unable to pull Docker image

### DISCUSSION THREAD

**Discussion by Kvaibhav01** [dailydotdev/daily#432] [General]
Hello!

I wanted to setup the [apps repo](https://github.com/dailydotdev/apps) locally. I have followed these steps:

1. Clone the repp.
2. Navigate to the apps folder, run the Docker Desktop app so that Docker services are running.
3. Run this command:

`docker-compose pull && docker-compose up`

It works well for some but doesn't work on others. Here's what I see on Terminal:

![image](https://user-images.githubusercontent.com/11731837/150184354-5facf684-387b-479b-823e-619b0f126d67.png)

And here's what the Docker Desktop app shows:

![image](https://user-images.githubusercontent.com/11731837/150184089-942f8d55-cc53-4fea-8ce0-0c03069f76e4.png)

**Comment by rebelchris** [comment]
Hi @Kvaibhav01 this might be related to an issue where we don't wait long enough for the MySQL to spool.
Could you try and see what happens if you run `docker-compose up` again?

If not please let me know and I'll try and help you more on this 👍

**Comment by Kvaibhav01** [comment]
Hey @rebelchris, I get the same issue when running `docker-compose up`


<img width="982" alt="CleanShot 2022-01-20 at 17 58 06@2x" src="https://user-images.githubusercontent.com/11731837/150338696-6985afe4-c716-44fe-94d9-73613fa4c2af.png">

**Reply by rebelchris** [reply]
Thanks for testing @Kvaibhav01 I've submitted a PR that should fix the issue you are having.

https://github.com/dailydotdev/apps/pull/853

**Reply by Kvaibhav01** [reply]
Hi! I see the PR is merged. Is it good to try again now?

**Reply by rebelchris** [reply]
Hi @Kvaibhav01 yes please pull the latest and re-run.
Let me know if it still has issues.

**Reply by Kvaibhav01** [reply]
I get the same issue as before. I recloned the apps directory and delete all the previous volumes.
![Screenshot of logs](https://user-images.githubusercontent.com/11731837/151383627-3aea25d9-48cc-40ed-ada1-9c326ae32970.png)

Also, I see these lines in between:

> daily-redis_1     | 1:M 27 Jan 2022 14:40:07.449 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.

> daily-postgres_1  | running bootstrap script ... 2022-01-27 14:40:07.663 UTC [28] FATAL:  could not open directory "pg_notify": Permission denied
daily-postgres_1  | child process exited with exit code 1
daily-postgres_1  | initdb: removing contents of data directory "/var/lib/postgresql/data"
daily-postgres_1  | could not open directory "/var/lib/postgresql/data": Permission denied
daily-postgres_1  | initdb: failed to remove contents of data directory
apps_daily-postgres_1 exited with code 1


> daily-mysql_1     | PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !

**Reply by rebelchris** [reply]
Hi @Kvaibhav01 there does seem to be an issue still with the Postgres database.
In our team we don't use this docker image, so sorry it's been a bit tricky to get it working again.

What seems to work for me, is to manually reboot the postgres if it fails, and then rerun `docker-compose pull && docker-compose up`

I'll keep looking at what causes this specifically, but it should help you get it up and running I think.
If not, are you on Discord? Perhaps we can do a live session which might be easier to find out how we can spool it up for you.

You can add me at: Chris Bongers#6139

**Reply by Kvaibhav01** [reply]
For me, I don't have the restart option in the Docker Desktop app. 

<img width="985" alt="Docker Desktop image" src="https://user-images.githubusercontent.com/11731837/151844893-5e553cff-3d69-45a0-9be2-cafbe192654d.png">

The command is still ruining with the same message:

> daily-api_1 | Operation timed out
apps_daily-api_1 exited with code 1

But when I click on the Start button on CLI these show up (along with more lines) randomly very frequently:

> daily-postgres_1  | 2022-01-31 17:39:48.313 UTC [5272] LOG:  database system was interrupted; last known up at 2022-01-31 17:39:48 UTC
daily-postgres_1  | 2022-01-31 17:39:48.338 UTC [5272] LOG:  database system was not properly shut down; automatic recovery in progress

I tried to re-run `docker-compose pull && docker-compose up` but no luck.

**Reply by Kvaibhav01** [reply]
Also, just sent you a request on Docker. Mine is: Kvaibhav01#4300
